app/models/host_facet_base.rb

```diff
@@ -0,0 +1,43 @@
+class HostFacetBase < ActiveRecord::Base
+  self.abstract_class = true
+
+  belongs_to_host
+
+  # Add facet's details to host's ENC by returning it from this method
+  # Basic ENC hash structure:
+  # ---
+  # classes:
+  #   ...
+  # parameters:
+  #   ...
+  # environment:
+  #   ...
+  def info
+    {}
+  end
+
+  # Specify any smart proxy id's used by this facet.
+  def smart_proxy_ids
```

Comment:
In `Host::Managed` there is already a [method](https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L743) `#smart_proxies`.